### PR TITLE
fix: duplicate download container images that already exist in base image

### PR DESCRIPTION
**Pull request by Lan-ce-lot**
Signed-off-by: Lancelot <1984737645@qq.com>

### Does this pull request fix one issue?
#1881 

### Describe how you did it

I use a way to find the Docker image locally whether the image has already existed in Base Image

### Describe how to verify it
Before
image

We don't need to download again
image

### CODE REVIEW

File: pkg/image/save/filesystem.go [hunk 1]
```diff
@@ -248,7 +248,9 @@ func (d *driver) Stat(ctx context.Context, subPath string) (storagedriver.FileIn
 func (d *driver) List(ctx context.Context, subPath string) ([]string, error) {
 	fullPath := d.fullPath(subPath)
 	// #nosec
-	dir, err := os.OpenFile(fullPath, os.O_WRONLY|os.O_CREATE, 0600)
+	// os.OpenFile for a file, but we need to read a dir
+	//dir, err := os.OpenFile(fullPath, os.O_WRONLY|os.O_CREATE, 0600)
```
[the remark this answers is not in the document]
Comment (Lan-ce-lot) [reply pkg/image/save/filesystem.go]: ok

**Comment by kakaZhou719** [comment]
nice work~~， if we find a way that only mount build layers without base image layers to do target image name search, then, named list will meet expectations . However , I also agree with the current approach.

**Comment by kakaZhou719** [comment]
> > nice work~~， if we find a way that only mount build layers without base image layers to do target image name search, then, named list will meet expectations . However , I also agree with the current approach.
> 
> Does this mean that we need a place to store the all container images, rather than in the base image?

as we know, a cluster image may contains those layers(base image layers+ build layers) , if we could only mount build layers to do search action except those base image layers, we will find the expected image list to download.

**Comment by Lan-ce-lot** [comment]
> as we know, a cluster image may contains those layers(base image layers+ build layers) , if we could only mount build layers to do search action except those base image layers, we will find the expected image list to download.

I got it, I will try your thoughts. :blush:

**Comment by VinceCui** [comment]
@Lan-ce-lot I know lan ce lot.

**Pull request merged**